python/pyarrow/_parquet_encryption.pyx

```diff
@@ -28,6 +28,12 @@ from pyarrow.lib cimport _Weakrefable
 from pyarrow.lib import tobytes, frombytes
 
 
+cdef extern from "Python.h":
+    # To let us get a PyObject* and avoid Cython auto-ref-counting
+    PyObject* PyBytes_FromStringAndSizeNative" PyBytes_FromStringAndSize"(
+        char *v, Py_ssize_t len) except NULL
```

Comment:
I'm a bit surprised that this declaration is required. Cython already provides [a declaration](https://github.com/cython/cython/blob/b862800bda6cc5f540d3b7961350cd2b6ad00dbd/Cython/Includes/cpython/bytes.pxd#L32-L36) for this function and it would be surprised if it didn't have the right ref-counting behavior.

(also, by the way, if we require Cython 3.1 we will gain implicit `std::string_view` conversion to/from `bytes` objects)

Reply:
Importing `PyBytes_FromStringAndSize` via `from cpython.bytes cimport PyBytes_FromStringAndSize` works.

I have copied this from `io.pxi`, maybe it is  not needed there as well / any more:
https://github.com/apache/arrow/blob/cdc14591248fcd7f6b6aa8005885a2dc817c4ccb/python/pyarrow/io.pxi#L43-L46